### DISCUSSION TITLE
How to compute the total length of a series of components

### DISCUSSION THREAD

**Discussion by tmerelle** [gdsfactory/gdsfactory#3376] [Q&A] [answered]
Hi there! 

I have a waveguide that is built up using a series of components that I connect using the .connect function.

Some are straight sections, some are cuved (euler) sections.

I'd like to compute the total effective length of the whole structure.

Thanks in advance for your help!

Thomas

**Comment by joamatab** [comment, marked as answer]
Hi Thomas,

We usually define derived parameters like length into the info of the PDK cells

```python
import gdsfactory as gf

c = gf.Component()
straight = gf.components.straight()
bend = gf.components.bend_circular()

straight_ref = c.add_ref(straight)
bend_ref = c.add_ref(bend)
bend_ref.connect("o1", straight_ref.ports["o2"])

length = straight.info['length'] + bend.info['length']
print(length) 

length = straight_ref.cell.info['length'] + bend_ref.cell.info['length']
print(length) 

c.show()

```

**Reply by niubiworker** [reply]
This answer is very good, but it uses the code `gf.components.coupler_ring(gap=0.3, radius=100, length_x=0, cross_section=gf.cross_section.strip(width=1.5))`. I only want to know the Euler bending length in this case. What should I do? Thank you very much!

**Comment by tmerelle** [comment]
Thanks Joaquin,

I figured it out in the meantime :)

Best regards,

Thomas

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
